### PR TITLE
Pre transaction modals explaining whats happening

### PR DESCRIPTION
**Pull request by tibetsprague**
Also showing the incentives to do each action

### CODE REVIEW

File: src/components/Proposal/PredictionBox.tsx [hunk 1]
```diff
@@ -158,6 +167,16 @@ export default class PredictionBox extends React.Component<IProps, IState> {
 
     return (
       <div className={wrapperClass}>
+        {showPreStakeModal ?
+          <PreTransactionModal
+            actionType={pendingPrediction == VoteOptions.Yes ? 'stakePass' : 'stakeFail'}
+            action={stakeProposal.bind(null, proposal.daoAvatarAddress, proposal.proposalId, stakeAmount, Number(stakeAmount))}
+            closeAction={this.closePreStakeModal.bind(this)}
+            dao={dao}
+            effectText={<span>Prediction amount: {stakeAmount} GENs<br/>You are predicting the proposal will be {pendingPrediction == VoteOptions.Yes ? "accepted" : "rejected"} by the DAO</span>}
```
Comment (dev-matan-tsuberi) [inline src/components/Proposal/PredictionBox.tsx]: "the" -> "that the"

**Comment by dev-matan-tsuberi** [comment]
Clicking "Launch Metamask" doesn't seem to actually do so. Not getting any popup. (tested on the review app)

File: src/reducers/arcReducer.ts [hunk 2]
```diff
@@ -184,6 +217,17 @@ export function proposalFailed(proposal: IProposalState) {
   return res;
 }
 
+export function proposalRewardsString(proposal: IProposalState) {
```
Comment (dev-matan-tsuberi) [inline src/reducers/arcReducer.ts]: seems like a UI thing, should not be in the reducer file.

File: src/components/Shared/PreTransactionModal.tsx [hunk 3]
```diff
@@ -0,0 +1,188 @@
...
+import { Modal } from 'react-router-modal';
+
+import * as arcActions from "actions/arcActions";
+import { IDaoState, IProposalState, ProposalStates, proposalEnded, proposalRewardsString } from "reducers/arcReducer";
+
+import * as css from "./PreTransactionModal.scss";
+
+interface IProps {
+  action: any;
+  actionType: string;
```
Comment (dev-matan-tsuberi) [inline src/components/Shared/PreTransactionModal.tsx]: could be an `enum`

File: src/components/Shared/PreTransactionModal.tsx [hunk 4]
```diff
@@ -0,0 +1,188 @@
...
+  }
+
+  public render() {
+    const { actionType, currentAccount, dao, effectText, proposal } = this.props;
+
+    const totalReputation = proposal.state == ProposalStates.Executed ? proposal.reputationWhenExecuted : dao.reputationCount;
+    const yesPercentage = totalReputation ? Math.round(proposal.votesYes / totalReputation * 100) : 0;
+    const noPercentage = totalReputation ? Math.round(proposal.votesNo / totalReputation * 100) : 0;
+
+    // TODO: fix
```
Comment (dev-matan-tsuberi) [inline src/components/Shared/PreTransactionModal.tsx]: what's there to fix?

File: src/components/Shared/PreTransactionModal.tsx [hunk 5]
```diff
@@ -0,0 +1,188 @@
...
+                  <div className={css.header}>
+                    <h2>Genesis Protocol</h2>
+                    <h3>RULES FOR YES VOTES</h3>
+                  </div>
+                  <div className={css.body}>
+                    { actionType == 'upvote' || actionType == 'downvote' ?
+                        <div>
+                          <p>When you vote, 1% of your reputation is taken away for the duration of the vote. If you vote for something to pass and it does, you gain reputation in addition to the reputation you lost.</p>
+                          <p>If you vote for something to pass and it does, you will be given a portion of whatever GEN have been staked on the proposal.</p>
+                          <p>If you vote for something to fail and it does, you will be given a portionof whatever GEN have been staked on the proposal.</p>
```
Comment (dev-matan-tsuberi) [inline src/components/Shared/PreTransactionModal.tsx]: "portionof" -> "portion of"

File: src/components/Proposal/VoteBox.tsx [hunk 6]
```diff
@@ -94,11 +102,22 @@ export default class VoteBox extends React.Component<IProps, IState> {
...
 
     return (
       <div className={wrapperClass}>
+        {this.state.showPreVoteModal ?
+          <PreTransactionModal
+            actionType={this.state.currentVote == 1 ? 'upvote' : 'downvote'}
+            action={voteOnProposal.bind(null, proposal.daoAvatarAddress, proposal, this.state.currentVote)}
+            closeAction={this.closePreVoteModal.bind(this)}
+            dao={dao}
+            effectText={<span>Your influence: <strong>{currentAccountReputationPercentage.toLocaleString(undefined, {minimumFractionDigits: 2, maximumFractionDigits: 2})}% ({currentAccountReputation.toLocaleString(undefined, {minimumFractionDigits: 2, maximumFractionDigits: 2})}) Reputation</strong></span>}
```
Comment (dev-matan-tsuberi) [inline src/components/Proposal/VoteBox.tsx]: Can use `<ReputationView/>`

File: src/components/Proposal/PredictionBox.tsx [hunk 7]
```diff
@@ -158,6 +167,16 @@ export default class PredictionBox extends React.Component<IProps, IState> {
 
     return (
       <div className={wrapperClass}>
+        {showPreStakeModal ?
+          <PreTransactionModal
```
Comment (dev-matan-tsuberi) [inline src/components/Proposal/PredictionBox.tsx]: Maybe should consider showing a bigger more explicit warning 
message in the modal if already have a sent staking tx in flight (`isStakingPass || isStakingFail`).

**Comment by dev-matan-tsuberi** [comment]
@tibetsprague sorry, my bad, working for me.

File: src/reducers/arcReducer.ts (hunk 2 above)
Comment (tibetsprague) [reply src/reducers/arcReducer.ts]: yeah, where should it go?

Comment (dev-matan-tsuberi) [reply src/reducers/arcReducer.ts]: `PreTransactionModal.tsx`

File: src/components/Proposal/PredictionBox.tsx (hunk 7 above)
Comment (tibetsprague) [reply src/components/Proposal/PredictionBox.tsx]: ill leave that up to Dor

**Comment by tibetsprague** [comment]
@dev-matan-tsuberi ready for another review

**Comment by tibetsprague** [comment]
@dev-matan-tsuberi unfortunately fixing that breaks the current layout, the text is too long

File: src/components/Proposal/PredictionBox.tsx (hunk 7 above)
Comment (dev-matan-tsuberi) [reply src/components/Proposal/PredictionBox.tsx]: Can we tag Dor on this? does he have a github?

Comment (tibetsprague) [reply src/components/Proposal/PredictionBox.tsx]: i created a ticket https://trello.com/c/dtRvcxoI

**Pull request merged**